src/Analyzers/Core/Analyzers/OrderModifiers/AbstractOrderModifiersDiagnosticAnalyzer.cs

```diff
@@ -71,8 +71,13 @@ private void AnalyzeSyntaxTree(SyntaxTreeAnalysisContext context, CompilationOpt
                 // If the severity is hidden, put the marker on all the modifiers so that the
                 // user can bring up the fix anywhere in the modifier list.
                 context.ReportDiagnostic(
-                    Diagnostic.Create(Descriptor, context.Tree.GetLocation(
-                        TextSpan.FromBounds(modifiers.First().SpanStart, modifiers.Last().Span.End))));
+                    DiagnosticHelper.Create(
```

Comment:
❗ The severity check a few lines up need to be corrected to reflect the severity that will be included in the custom diagnostic.

Reply:
What is the expected behavior for this check in the first place? From my testing 8.0.1xx and this PR behave in the same way.
If by correcting you mean it actually checking the final severity and not just the option custom severity then I'd say it's out of scope of this PR, and I'm not sure whether it's even possible here.

Reply:
Ah, makes sense, thanks.